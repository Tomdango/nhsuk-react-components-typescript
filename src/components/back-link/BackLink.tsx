import React, { HTMLProps } from 'react';
import classNames from 'classnames';
import { ChevronLeft } from '../icons';

const BackLink: React.FC<HTMLProps<HTMLAnchorElement>> = ({ children, className, ...rest }) => (
  <div className="nhsuk-back-link">
    <a className={classNames('nhsuk-back-link__link', className)} {...rest}>
      <ChevronLeft />
      {children}
    </a>
  </div>
);

export default BackLink;
