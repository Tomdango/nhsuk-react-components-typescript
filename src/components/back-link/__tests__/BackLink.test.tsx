import React from 'react';
import BackLink from '..';
import { shallow } from 'enzyme';

describe('BackLink', () => {
  it('matches snapshot', () => {
    const component = shallow(<BackLink href="/back">Back</BackLink>);
    expect(component).toMatchSnapshot('BackLink');
    component.unmount();
  });
});
