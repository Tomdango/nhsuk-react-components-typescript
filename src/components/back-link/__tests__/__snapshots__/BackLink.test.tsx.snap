// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackLink matches snapshot: BackLink 1`] = `
<div
  className="nhsuk-back-link"
>
  <a
    className="nhsuk-back-link__link"
    href="/back"
  >
    <Component />
    Back
  </a>
</div>
`;
